packages/channel/src/data-refresh.ts

```diff
@@ -59,6 +60,19 @@ export const refreshData = async ({
       await refreshDataOf(integration, initial);
       deleteInsightsCache(integration.organizationId);
     }
+
+    const organizationIds = new Set(integrations.map((integration) => integration.organizationId));
+    for (const organizationId of organizationIds) {
+      await getInsightsHelper(
+        {
+          interval: 'week',
+          orderBy: 'spend_abs',
+          page: 1,
+          pageSize: 3,
```

Comment:
```suggestion
          pageSize: 3,
          groupBy: ['integrationType']
```
if we don't group my anything we will have the 3 top of the thousands insights of this org. If you group you get the top 3 out of the number of connected integrations.